content/client-lib-development-guide/features.textile

```diff
@@ -276,8 +276,11 @@ h3(#realtimeclient). RealtimeClient
 * @(RTC6)@ @RealtimeClient#time@ function:
 ** @(RTC6a)@ Proxy to @RestClient#time@ presented with an async or threaded interface as appropriate
 * @(RTC7)@ The client library must use the configured timeouts specified in the @ClientOptions@, falling back to the "client library defaults":#defaults and defaults described in @ClientOptions@ below
-* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth.authorise@ is called with @AuthOptions#force@ set to @true@ and the state is @CONNECTED@, @CONNECTING@, or @DISCONNECTED@, the client will obtain a new token, disconnect the current transport and resume the connection (see "@RTN15c@":#RTN15c for details on resume). A test should exist that performs an upgrade of capabilities without any message loss during the upgrade process. A test should also exist for a token change that fails due to an incompatible token, which should result in the connection entering the @FAILED@ state
-* @(RTC9)@ @RealtimeClient#request@ is a wrapper around @RestClient#request@ (see "RSC19":#RSC19) delivered in an idiomatic way for the realtime library i.e. in the case of Ruby, with an evented async callback interface
+* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth#authorize@ is called or Ably requests a re-authentication (see "RTN22":#RTN22), the client must obtain a new token and send an @AUTH@ @ProtocolMessage@ to Ably. The @auth@ attribute of the @ProtocolMessage@ must contain a JSON-like object in the format @{ "accessToken": "<token string>" }@ or @{ "tokenDetails": <TokenDetails> }@
```

Comment:
Lgtm

On 11 Aug 2016 9:58 a.m., "mattheworiordan" <notifications@github.com>
wrote:

> In content/client-lib-development-guide/features.textile
> <https://github.com/ably/docs/pull/132#discussion_r74380817>:
>
> > @@ -276,8 +276,11 @@ h3(#realtimeclient). RealtimeClient
> >  * @(RTC6)@ @RealtimeClient#time@ function:
> >  ** @(RTC6a)@ Proxy to @RestClient#time@ presented with an async or threaded interface as appropriate
> >  * @(RTC7)@ The client library must use the configured timeouts specified in the @ClientOptions@, falling back to the "client library defaults":#defaults and defaults described in @ClientOptions@ below
> > -* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth.authorise@ is called with @AuthOptions#force@ set to @true@ and the state is @CONNECTED@, @CONNECTING@, or @DISCONNECTED@, the client will obtain a new token, disconnect the current transport and resume the connection (see "@RTN15c@":#RTN15c for details on resume). A test should exist that performs an upgrade of capabilities without any message loss during the upgrade process. A test should also exist for a token change that fails due to an incompatible token, which should result in the connection entering the @FAILED@ state
> > -* @(RTC9)@ @RealtimeClient#request@ is a wrapper around @RestClient#request@ (see "RSC19":#RSC19) delivered in an idiomatic way for the realtime library i.e. in the case of Ruby, with an evented async callback interface
> > +* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth#authorize@ is called or Ably requests a re-authentication (see "RTN22":#RTN22), the client must obtain a new token and send an @AUTH@ @ProtocolMessage@ to Ably. The @auth@ attribute of the @ProtocolMessage@ must contain a JSON-like object in the format @{ "accessToken": "<token string>" }@ or @{ "tokenDetails": <TokenDetails> }@
>
> @paddybyers <https://github.com/paddybyers> what about bd2ed74
> <https://github.com/ably/docs/commit/bd2ed744f6048b9b36c60aa7625363845c7838f4>
> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ably/docs/pull/132/files/5e1f00061c8622d0d655be33de5583edeb588bf4#r74380817>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABFbpvlCzB7mV2Bnk4ynSFx7BBw8BeINks5qetYcgaJpZM4JUNjA>
> .
>